tests/integration/test_charm.py

```diff
@@ -99,3 +117,57 @@ async def test_application_config(ops_test: OpsTest, get_unit_ip_list) -> None:
             response = requests.get(f"http://{unit_ip}:8080/hello-world", timeout=5)
             assert response.status_code == 200
             assert "Bonjour" in response.text
+
+
+async def test_jvm_config(ops_test: OpsTest, get_unit_ip_list) -> None:
+    """
+    arrange: deploy Spring Boot applications.
+    act: Update the jvm-config to configure JVM heap memory settings.
+    assert: Spring Boot example applications should run with corresponding JVM parameters.
+    """
+    assert ops_test.model
+    jvm_config = "-Xms256m -Xmx512M"
+    await asyncio.gather(
+        ops_test.model.applications[APP_NAME].set_config({"jvm-config": jvm_config}),
+        ops_test.model.applications[BUILDPACK_APP_NAME].set_config({"jvm-config": jvm_config}),
+        ops_test.model.wait_for_idle(apps=[APP_NAME, BUILDPACK_APP_NAME], status="active"),
+    )
+    for name in [APP_NAME, BUILDPACK_APP_NAME]:
+        unit_ips = await get_unit_ip_list(name)
+        for unit_ip in unit_ips:
+            response = requests.get(f"http://{unit_ip}:8080/jvm-arguments", timeout=5)
+            assert response.status_code == 200
+            assert response.json() == jvm_config.split()
+
+
+async def test_invalid_jvm_config(ops_test: OpsTest) -> None:
+    """
+    arrange: deploy Spring Boot applications.
+    act: Update the jvm-config with some invalid values.
+    assert: Spring Boot example applications should enter blocked status.
+    """
+    assert ops_test.model
+    invalid_jvm_config = "-Xms256A -Xmx512G"
+    await asyncio.gather(
+        ops_test.model.applications[APP_NAME].set_config({"jvm-config": invalid_jvm_config}),
+        ops_test.model.applications[BUILDPACK_APP_NAME].set_config(
+            {"jvm-config": invalid_jvm_config}
+        ),
+        ops_test.model.applications[MEM_1G_APP_NAME].set_config({"jvm-config": "-Xms256m -Xmx2G"}),
+        ops_test.model.applications[MEM_1G_BUILDPACK_APP_NAME].set_config(
+            {"jvm-config": "-Xms256m -Xmx2G"}
+        ),
+        ops_test.model.wait_for_idle(apps=ALL_APP_NAMES),
+    )
+    for name in [APP_NAME, BUILDPACK_APP_NAME]:
+        for unit in ops_test.model.applications[name].units:
+            assert unit.workload_status == "blocked"
+            assert unit.workload_status_message == "Invalid jvm-config"
+
+    for name in [MEM_1G_APP_NAME, MEM_1G_BUILDPACK_APP_NAME]:
```

Comment:
Similar comment on tuple and `all` function

Reply:
Same as the comment about `next` above.